Signal/src/ViewControllers/AddContactShareToExistingContactViewController.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import ContactsUI
+
+class AddContactShareToExistingContactViewController: ContactsPicker, ContactsPickerDelegate, CNContactViewControllerDelegate {
+
+    // TODO - there are some hard coded assumptions in this VC that assume we are *pushed* onto a
+    // navigation controller. That seems fine for now, but if we need to be presented as a modal,
+    // or need to notify our presenter about our dismisall or other contact actions, a delegate
+    // would be helpful. It seems like this would require some broad changes to the ContactShareViewHelper,
+    // so I've left it as is for now, since it happens to work.
```

Comment:
I agree - for any non-trivial VC (e.g. VCs with delegates), it'd be nice to let the delegate handle dismissal.

Reply:
What I mean is that we should be building view controllers such that adding a delegate is easy. To the point - in this case I couldn't easily do it because the presentation of this VC occurs as a result of a nested series of calls in a view helper class.

The solution isn't totally obvious to me - but I think the problem is: "What's a good way to extract view controller logic so it can be composed without inheritance".